rust/lance-core/src/datatypes/schema.rs

```diff
@@ -44,8 +44,7 @@ struct SchemaFieldIterPreOrder<'a> {
 impl<'a> SchemaFieldIterPreOrder<'a> {
     #[allow(dead_code)]
     fn new(schema: &'a Schema) -> Self {
-        let mut field_stack = Vec::new();
-        field_stack.reserve(schema.fields.len() * 2);
+        let mut field_stack = Vec::with_capacity(schema.fields.len() * 2);
```

Comment:
This is unrelated but clippy keeps warning about it on my system so I just threw it in